velox/docs/develop/simd.rst

```diff
@@ -0,0 +1,179 @@
+===================
+SIMD Usage in Velox
+===================
+
+SIMD uses special registers in CPU to operate on multiple primitive data at same
+time.  In some basic cases compiler is able to translate a tight loop into SIMD
+instructions for us, but in some cases we need to call the SIMD intrinsics
+explicitly.
+
+There are several places in Velox where we use SIMD explicitly to get better
+performance.  We also use ``xsimd`` to build zero-cost abstraction over the
+intrinsics, to address portability issue.
+
+Architectures
+-------------
+
+In Velox we support 2 families of CPU architectures regarding SIMD: X86 and ARM.
+In X86 there are 3 generations of SIMD technologies: SSE, AVX, and AVX512.  For
+ARM there are NEON and SVE.  Each architecture has its own size of registers, we
+summarize the details below:
+
+============ ==================== ==================
+Architecture Register Size (bits) Supported in Velox
+============ ==================== ==================
+SSE          128                  Yes
+AVX          256                  Yes
+AVX512       512                  No
+NEON         128                  Yes
+SVE          128 - 2048           No
+============ ==================== ==================
+
+xsimd Basics
```

Comment:
xsimd -> XSimd ?

Reply:
Not sure how they would capitalize the name in a title; they pretty much using all lower case exclusively.